searchlib/src/vespa/searchlib/common/allocatedbitvector.cpp

```diff
@@ -70,41 +70,38 @@ AllocatedBitVector::AllocatedBitVector(Index numberOfElements, Alloc buffer, siz
     }
 }
 
-AllocatedBitVector::AllocatedBitVector(Index numberOfElements, Index capacityBits, const void * rhsBuf, size_t rhsSize, const Alloc* init_alloc) :
-    BitVector(),
-    _capacityBits(capacityBits),
-    _alloc(allocatePaddedAndAligned(0, numberOfElements, capacityBits, init_alloc))
+AllocatedBitVector::AllocatedBitVector(Index numberOfElements, Index capacityBits, const BitVector* org, const Alloc* init_alloc)
+    : BitVector(),
+      _capacityBits(capacityBits),
+      _alloc(allocatePaddedAndAligned(0, numberOfElements, capacityBits, init_alloc))
 {
     _capacityBits = computeCapacity(_capacityBits, _alloc.size());
     init(_alloc.get(), 0, numberOfElements);
-    clear();
-    if (rhsSize > 0) {
-        size_t minCount = std::min(static_cast<size_t>(numberOfElements), rhsSize);
-        memcpy(getStart(), rhsBuf, numBytes(minCount));
-        if (minCount/8 == numberOfElements/8) {
-            static_cast<Word *>(getStart())[numWords()-1] &= ~endBits(minCount);
-        }
-        set_bit_no_range_check(size()); // Guard bit
+    if (org != nullptr) {
+        initialize_from(*org);
+        set_bit_no_range_check(size());
```

Comment:
I think it could be useful to preserve the guard bit comment from the previous version here